src/MatrixTwitter.js

```diff
@@ -0,0 +1,987 @@
+var Twitter = require('twitter');
+var Request = require('request');
+var fs = require('fs');
+var log = require('npmlog');
+var Buffer = require('buffer').Buffer;
+var HTMLDecoder = new require('html-entities').AllHtmlEntities;
+var MatrixRoom = require("matrix-appservice-bridge").MatrixRoom;
+var RemoteRoom = require("matrix-appservice-bridge").RemoteRoom;
+
+var ProcessedTweetList = require("./ProcessedTweetList.js");
+var util = require("./util.js");
+
+const TWITTER_PROFILE_INTERVAL_MS   = 300000;
+const TWITTER_CLIENT_INTERVAL_MS    = 60000;
+const TWITTER_MSG_QUEUE_INTERVAL_MS = 1500;
+const TWEET_REPLY_MAX_DEPTH = 3;
+
+
+/**
+ * MatrixTwitter - This class handles the connections between the Twitter API
+ * and the bridge.
+ * @class
+ * @param  {matrix-appservice-bridge.Bridge} bridge
+ * @param  config. Bridge configuration (currently only make use of the auth
+*  information)
+ * @param  config.app_auth.consumer_key Twitter consumer key
+ * @param  config.app_auth.consumer_secret Twitter consumer secret
+ * @param  {TwitterDB}       storage
+ */
+var MatrixTwitter = function (bridge, config, storage) {
+  this.app_auth = config.app_auth;
+  this.app_twitter = null;
+  this.tclients = {};
```

Comment:
Don't worry, [I'm like this with everyone](https://github.com/matrix-org/matrix-appservice-gitter/pull/8#discussion_r69431126).